nilearn/glm/first_level/experimental_paradigm.py

```diff
@@ -86,14 +87,13 @@ def check_events(events):
     unexpected_columns = set(events_copy.columns).difference(VALID_FIELDS)
     for unexpected_column in unexpected_columns:
         warnings.warn(("Unexpected column `{}` in events "
-                       "data will be ignored.").format(
-                            unexpected_column))
+                       "data will be ignored.").format(unexpected_column))
 
     # Make sure we have a numeric type for duration
     if not is_numeric_dtype(events_copy['duration']):
         try:
             events_copy = events_copy.astype({'duration': float})
-        except:
+        except ValueError:
             raise ValueError("Could not cast duration to float "
                              "in events data.")
```

Comment:
I haven't checked this. Is the original error definitely a ValueError?

Reply:
I believe so because this test still passes after my change, but maybe there are other edge cases not tested?

https://github.com/nilearn/nilearn/blob/4555aaaa78e9baff7a2ff727c7ea30a98b753f00/nilearn/glm/tests/test_paradigm.py#L97-L101